io/StacReader.cpp

```diff
@@ -578,11 +595,11 @@ void StacReader::initialize()
 
     std::string stacType = stacJson.at("type");
     if (stacType == "Feature")
-        initializeItem(stacJson);
+        initializeItem(stacJson, m_filename);
```

Comment:
These are member functions that are always going to have access to `m_filename`. Rather than adding arguments to functions, maybe it would be better to add to the `NL::json` any special members that you might possibly need. As it is now, it is confusing - why do I need both a path and a json member and what does the path need to be?

Reply:
`m_filename` in this instance is the src path that everything is being related to. I could maybe add a `relative_to` key to the STAC JSON there? I don't know if that feels any better from a programmatic standpoint though